ui/app/decorators/unload-model-route.js

```diff
@@ -0,0 +1,48 @@
+import { inject as service } from '@ember/service';
+import { removeModel } from '../utils/remove-record';
+
+/**
+ * Takes array of model names which should be unloaded when moving away from
+ * the route. Automatically unloads all `capabilities` and `control-group`
+ * data before unloading the specified models.
+ *
+ *** basic example
+ *
+ * import Route from '@ember/route';
+ * import withUnloadModelRoute from 'vault/decorators/unload-model-route';
+ *
+ * @withUnloadModelRoute(['policy/acl'])
+ * class PolicyViewRoute extends Route { foo = null; }
+ *
+ */
+
+export function withUnloadModelRoute(models) {
+  return function decorator(SuperClass) {
+    return class UnloadModelRoute extends SuperClass {
+      @service store;
+
+      static _models;
```

Comment:
What if instead of setting the models to `_models` on the extended class, we access the `models` argument directly in `unloadModels` and `deactivate`?